utils/types.go

```diff
@@ -156,10 +131,15 @@ func GetKongClient(opt KongClientConfig) (*kong.Client, error) {
 	defaultTransport.TLSClientConfig = &tlsConfig
 	c.Transport = defaultTransport
 	if len(opt.Headers) > 0 {
```

Comment:
This `if` seems superfluous at first glance. To keep code simple (and easier to test), could we always run with headers, even if the header list is empty?

Reply:
Low level performance is the reason.
The implementation of the the RoundTrip will do unnecessary copies: https://github.com/Kong/go-kong/blob/main/kong/utils.go#L64-L75

In your opinion, what is the right place to have this guard? Inside RoundTrip or here? And why?
(I think inside the RoundTrip. Why? Make life easy for the caller. The caller shouldn't even know anything about the implementation.)

Reply:
I think that the place for such optimizations is under the abstraction.

That said, I think that's too small of a detail to block this PR.